client/state/sites/plans/assembler.js

```diff
@@ -9,6 +9,7 @@ const createSitePlanObject = ( plan ) => {
 	}
 
 	return {
+		auto_renew: Boolean( plan.auto_renew ),
```

Comment:
Oh if we're bothering to convert this in assembler.js, we might as well change this to 🐫 case to match the other props within this object.